pkg/serdes/avro_deserialization_provider.go

```diff
@@ -44,6 +44,9 @@ func (a *AvroDeserializationProvider) InitDeserializer(srClientUrl, srClusterId,
 	}
 
 	serdeConfig := avrov2.NewDeserializerConfig()
+	serdeConfig.RuleConfig = map[string]string{
```

Comment:
This should be removed.  It is only for testing the local KMS client.  If you need this for your tests, you should find a way to pass it in from the test.

Alternatively in your test you can set an environment var of the form
```
EXPORT LOCAL_SECRET=avro-secret
```

Reply:
Okay, sounds good, I will leave this as a cleaning up item in the end after finishing the main issue here regarding PROTOBUF, please refer to my latest reply in our channel.